lib/app/interactive/interactive.ml

```diff
@@ -159,11 +159,35 @@ let apply_patch hunk_patch =
   (if debug then Lwt_io.printf "[debug] %s,%s\n" stdout stderr else return ())
   >>= fun () -> process#close
 
+let apply_patch_with_git hunk_patch =
+  apply_patch_with_cmd "git apply --index --intent-to-add" hunk_patch
+
+let apply_patch hunk_patch =
+  apply_patch_with_cmd "patch -p 0" hunk_patch
+
 let drop_into_editor editor path ~at_line =
   let command = Format.sprintf "%s +%d %s" editor at_line path in
   Lwt_unix.system command
 
+let file_in_git_repo path =
+  let command = Format.sprintf "test ! -z \"$(git ls-files -- %s)\"" path in
+  Lwt_io.printf "[debug] %s\n" command |> ignore;
+  Lwt_unix.system command
+  >>= fun status ->
+  match status with
+  | Lwt_unix.WEXITED x -> Lwt_io.printf "[debug] %d\n" x |> ignore; return (x == 0)
+  | _ -> return false
```

Comment:
@rvantonder is it too much trouble to do this check?
It still not complete, need to check if the command is ok (if the git command fails because it doesn't exist or something...).

Reply:
This should be ok.

Reply:
seems reasonable! does this command work for all kinds of shells? `bash` / `sh` / `zsh`?

Reply:
`test` is from coreutils, so it must be everywhere, except windows, maybe...